msm-sitemap.php

```diff
@@ -570,24 +571,50 @@ public static function create_post_type() {
 			);
 	}
 
+	/**
+	 * Record deleted posts in an option to be used by get_last_modified_posts.
+	 */
+	public static function record_sitemap_to_rebuild( $post_id ) {
+		$deleted_posts = get_option( 'msm_sitemap_deleted_posts', array() );
+		$post = get_post( $post_id );
+		if ( ! $post ) {
+			return;
+		}
+
+		if ( in_array( $post->post_type, self::get_supported_post_types() ) ) {
+			$deleted_posts[] = array(
+				'ID' => (string) $post->ID,
+				'post_date' => $post->post_date,
+				);
+			update_option( 'msm_sitemap_deleted_posts', $deleted_posts, false );
+		}
+	}
+
 	/**
 	 * Get posts modified within the last hour
 	 * @return object[] modified posts
 	 */
-	public static function get_last_modified_posts() {
+	public static function get_last_modified_posts( $start_time = null) {
 		global $wpdb;
 
+		if ( ! $start_time ) {
+			$start_time = current_time( 'timestamp', 1 );
+		}
+
 		$sitemap_last_run = get_option( 'msm_sitemap_update_last_run', false );
 
-		$date = date( 'Y-m-d H:i:s', ( current_time( 'timestamp', 1 ) - 3600 ) ); // posts changed within the last hour
+		$date = date( 'Y-m-d H:i:s', ( $start_time - 3600 ) ); // posts changed within the last hour
 
 		if ( $sitemap_last_run ) {
 			$date = date( 'Y-m-d H:i:s', $sitemap_last_run );
 		}
 
 		$post_types_in = self::get_supported_post_types_in();
 
-		$modified_posts = $wpdb->get_results( $wpdb->prepare( "SELECT ID, post_date FROM $wpdb->posts WHERE post_type IN ( {$post_types_in} ) AND post_modified_gmt >= %s LIMIT 1000", $date ) );
+		$modified_posts = $wpdb->get_results( $wpdb->prepare( "SELECT ID, post_date FROM $wpdb->posts WHERE post_type IN ( {$post_types_in} ) AND post_modified_gmt >= %s LIMIT 1000", $date ), ARRAY_A );
```

Comment:
Would changing this to ARRAY_A cause problems with https://github.com/Automattic/msm-sitemap/blob/d4b04ccbb594f60245ef153295107e3b63a120b8/msm-sitemap.php#L602 where, from what I understand, it's assumed that $post->post_date will work?